### PR TITLE
Revert "Revert Merge pull request #586 ... "

### PR DESCRIPTION
**Pull request by lsebrie**
This reverts commit 468a1f963fdf41eb23e631e674c76c1089390583.
This commit brings back changes to integrate for REMASC.
The changes included here won't require a hard fork

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 2 issues

1. ![MAJOR][MAJOR] [Remasc.java#L160](https://github.com/rsksmart/rskj/blob/232d5c757b4a4bc4833988e88ac1c3516473d73d/rskj-core/src/main/java/co/rsk/remasc/Remasc.java#L160): Method co.rsk.remasc.Remasc.processMinersFees() accesses list or array with constant index [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ACLI_CONSTANT_LIST_INDEX)
1. ![INFO][INFO] [Remasc.java#L213](https://github.com/rsksmart/rskj/blob/232d5c757b4a4bc4833988e88ac1c3516473d73d/rskj-core/src/main/java/co/rsk/remasc/Remasc.java#L213): Redundant nullcheck of uncles, which is known to be non-null in co.rsk.remasc.Remasc.addNewSiblings() [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3ARCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by donequis** [comment]
The possibility of a fork is the reason why we can't.

There's an idea of having a nice data structure to access these blocks which will allow O(1) for best chain blocks and O(log n) for forks, but it was a separated issue

File: rskj-core/src/main/java/co/rsk/remasc/Remasc.java [hunk 1]
```diff
@@ -92,17 +94,38 @@ void processMinersFees() throws IOException, BlockStoreException {
...
-
         long processingBlockNumber = blockNbr - remascConstants.getMaturity();
         if (processingBlockNumber < 1 ) {
             logger.debug("First block has not reached maturity yet, current block is {}", blockNbr);
             return;
         }
 
-        Block processingBlock = blockStore.getBlockByHashAndDepth(executionBlock.getParentHash().getBytes(), remascConstants.getMaturity() - 1);
+        int uncleGenerationLimit = config.getBlockchainConfig().getCommonConstants().getUncleGenerationLimit();
+        List<Block> descendantsBlocks = new ArrayList<>(uncleGenerationLimit);
```
Comment (donequis) [inline rskj-core/src/main/java/co/rsk/remasc/Remasc.java]: Lets change this for LinkedLista + .push(·)

**Comment by lsebrie** [comment]
This PR is closed since another one contains all these changes are more progress

**Pull request closed without merging**